library/kotlin/src/io/envoyproxy/envoymobile/Client.kt

```diff
@@ -0,0 +1,73 @@
+package io.envoyproxy.envoymobile
+
+import java.nio.ByteBuffer
+
+interface Client {
+  fun startStream(request: Request, streamCallback: StreamCallback): StreamEmitter
+}
+
+interface StreamCallback {
+  fun onHeaders(statusCode: Int, headers: Map<String, List<String>>)
+  fun onData(byteBuffer: ByteBuffer, endStream: Boolean)
+  fun onMetadata(metadata: Map<String, List<String>>, endStream: Boolean)
+  fun onTrailers(trailers: Map<String, List<String>>)
+  fun onError(error: Error)
+
+  fun onCanceled() // TODO: onInterrupt()?
+  fun onCompletion()
+
+}
+
+interface StreamEmitter {
+  fun sendHeaders(headers: Map<String, List<String>>): Boolean
```

Comment:
These don't really align with the request interfaces checked in upstream - is the intent for public consumers not to use these interfaces, but to rather have them be used within the library for calling the C layer? For example, I'd expect an end consumer to be able to pass a `Request` model (with a URL, body, etc.) rather than having to call each of these functions to perform an HTTP request

Reply:
Per offline discussion, we should remove this, using:
- The `Request` for headers
- `sendMetadata`
- `sendTrailers`